### ISSUE TITLE
Starting ghci with a different command

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/atom-tidalcycles#99]
I'm experimenting with nix as a reliable way to install tidal https://github.com/tidalcycles/Tidal/issues/637

As a result I now want to start ghci like this:
```
nix-shell -p "haskellPackages.ghcWithPackages (pkgs: [pkgs.tidal])" --run ghci
```
However if I try using this as the ghci setting in the atom plugin I get this when I boot tidal:
```
/bin/sh: 1: Syntax error: word unexpected (expecting ")")
```
I've tried escaping spaces etc with `\` but no luck.

**Comment by ndr-brt** [comment]
There are problems with the handling of "ghci paths with whitespaces"
For handling that we put double quotes around path as here:
https://github.com/tidalcycles/atom-tidalcycles/blob/25b0c9a79fe7b9b30840dc1868852ee5464f93cb/lib/ghc.js#L37
and here:
https://github.com/tidalcycles/atom-tidalcycles/blob/25b0c9a79fe7b9b30840dc1868852ee5464f93cb/lib/ghc.js#L50

will look for a workaround

**Comment by ndr-brt** [comment]
also... this returns error:
`nix-shell -p "haskellPackages.ghcWithPackages (pkgs: [pkgs.tidal])" --run ghc-pkg field tidal data-dir`

**Comment by yaxu** [comment]
You'd need to also install nix, and run `nix-env -iA nixpkgs.haskellPackages.tidal_1_4_9` to install tidal within that.

Maybe we just need a 'use custom interpreter command' override, or something like that.

**Comment by ndr-brt** [comment]
I have nix installed, and this command works:
`nix-shell -p "haskellPackages.ghcWithPackages (pkgs: [pkgs.tidal])" --run ghci`

but the one with ghc-pkg does not...

**Comment by yaxu** [comment]
Ah, I see. This works for me:
```
nix-shell -p "haskellPackages.ghcWithPackages (pkgs: [pkgs.tidal])" --run "ghc-pkg field tidal data-dir"
```

**Comment by ndr-brt** [comment]
interesting... 
there's some work to do to handle all the cases, but, we can do it!

**Comment by ndr-brt** [comment]
Fixed in 3.9.0

**Comment by yaxu** [comment]
Thanks a lot! I'll try it out
